src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -240,7 +242,18 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuin
 
             // Read the first vector unaligned.
 
-            currentMask = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
+            if (Sse2.IsSupported)
+            {
+                currentMask = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
+            }
+            else if (AdvSimd.Arm64.IsSupported)
+            {
+                currentMask = Unicode.Utf16Utility.GetNonAsciiBytes(AdvSimd.LoadVector128(pBuffer), bitmask); // unaligned load
```

Comment:
Is this correct? You want the first non ascii byte in the `pBuffer`. You want to call https://github.com/dotnet/runtime/blob/cd8759d1bc94778f0bf35bc99dcdabf1b40cd71c/src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/AdvSimdHelper.cs#L114 to get that.

Reply:
I hope that some test case fails because of this. If not, you might want to add test coverage for it to catch it.

Reply:
Tests were failing because of this. Your comment helped track it down much faster than my print statements. I believe this is fixed now.